### PR TITLE
Make test_gestalt use SimulatedBackend

### PR DESCRIPTION
**Pull request by dcolligan**
We should test some kind of return values from the server before this is committed (as per the issue).

Issue #226

### CODE REVIEW

**Comment by dcolligan** [comment]
@jeromekelleher this should work as specified in the issue now

**Comment by jeromekelleher** [comment]
It would be nice to make the config parameterisable, so that we're not depending on the values embedded in test_server.py. I'm not sure this is a good name either -- do we run any tests in `test_server.py`? If not, we should just rename it `server.py`, or just move all of the functionality into `test_gestalt`, which seems to be the only place it's used.

**Comment by dcolligan** [comment]
The config is parameterisable.  That's what `ServerTestConfigFile` is for.

I would advocate keeping the server startup stuff in a separate module, since I could definitely imagine this being used by other end to end tests in the future.

File: tests/end_to_end/test_server.py [hunk 1]
```diff
@@ -112,3 +107,52 @@ def getServerErrLines(self):
...
+
+
+class ServerTestConfigFileDeterministic(ServerTestConfigFile):
+    """
+    Launch a server test with a custom configuration file
+    that ensures there is no pseudorandom behavior.
+    """
+    def setConfig(self):
+        self.config = """
+SIMULATED_BACKEND_NUM_VARIANT_SETS = 1
```
Comment (jeromekelleher) [inline tests/end_to_end/test_server.py]: This doesn't look very configurable to me - what if I want 5 variantSets?

**Comment by jeromekelleher** [comment]
OK, I won't argue about keeping the server stuff in a separate module, I think we should just rename it. It shouldn't be picked up and run by nosetests unless it contains actual test cases (which I think it doesn't?).

File: tests/end_to_end/test_server.py (hunk 1 above)
Comment (dcolligan) [inline tests/end_to_end/test_server.py]: Then you override `ServerTestConfigFile` and implement `setConfig` in your subclass with 5 variantsets and whatever else you want in the config.

**Comment by dcolligan** [comment]
@jeromekelleher pushed a change with a rename

File: tests/end_to_end/test_server.py (hunk 1 above)
Comment (jeromekelleher) [inline tests/end_to_end/test_server.py]: What about just adding some parameters to setConfig here? numVariantSets=1 by default? I don't want to have to reproduce this exact code just to change stuff like that.

Also, we should have more than one variant set to test the client properly, and I'm dead against the tests depending on a value embedded within a string in a different module.

Comment (dcolligan) [inline tests/end_to_end/test_server.py]: I'll move that code to the `test_gestalt.py` file then.

**Comment by dcolligan** [comment]
@jeromekelleher pushed a change that moved some code around

**Comment by jeromekelleher** [comment]
Excellent, thanks. +1.

**Pull request merged**